### ISSUE TITLE
Challenges with restore

### ISSUE THREAD

**Issue by CatmanV2** [bwssytems/ha-bridge#1221]
Hi
HA Bridge is fantastic. One thing though that I'd like to understand is the restore.
I recently had cause to totally rebuild my system, Thinking it would be comparatively simple I backed up the HA bridge devices, but they simply would not restore to the new system.

Errors were 500 and 400 and User not found.  I tried everything I could think of including restoring the .bk file, opening the restored .db file and copying the contents manually into the 'working' db file. Played with all the permissions I could think of as well.

Issue solved now, I rebuilt everything from scratch, but would like to know how I might avoid this in future.

This was the latest master of HA bridge on Debian Buster.

Cheers

C

**Comment by bwssytems** [comment]
Will need to take a looj

**Comment by dougdalton** [comment]
I have had the same problem,  I get error 400 with every attempt to restore

**Comment by bwssytems** [comment]
I seem to not be able to recreate this. Will need exact steps and your logs to research.

**Comment by CatmanV2** [comment]
If we attempt a restore, is there not a risk that we’ll wipe out our current set up? Otherwise I’m happy to try

C
-- 
Chris Matthews
chris@systemlabs.co.uk <mailto:chris@systemlabs.co.uk>
https://www.linkedin.com/in/cmatthewsitdirector/ <https://www.linkedin.com/in/cmatthewsitdirector/>
+447964 207823 / +447855 823629

**Comment by bwssytems** [comment]
Just take your own copies of the device.db and the habridge.config in your data directory and put them in a safe place. Then try to do backup and restores.

**Comment by CatmanV2** [comment]
OK that’s the theory.

Again not bing difficult but when I tried to copy the device.db manually into a new install, it completely bombed out.  Can’t recall the exact details, it was some months back :( I *only* copied the .db file though. Is the .config a pre-requisite as well?

Cheers!

C
-- 
Chris Matthews
chris@systemlabs.co.uk <mailto:chris@systemlabs.co.uk>
https://www.linkedin.com/in/cmatthewsitdirector/ <https://www.linkedin.com/in/cmatthewsitdirector/>
+447964 207823 / +447855 823629

**Comment by bwssytems** [comment]
Yes, you needed to copy the habridge.config as well as that has all the control settings.

**Comment by CatmanV2** [comment]
Got it. Thanks.

If I’m feeling brave I’ll try again!

C
-- 
Chris Matthews
chris@systemlabs.co.uk <mailto:chris@systemlabs.co.uk>
https://www.linkedin.com/in/cmatthewsitdirector/ <https://www.linkedin.com/in/cmatthewsitdirector/>
+447964 207823 / +447855 823629

**Comment by bwssytems** [comment]
Just copy those 2 files to another directory and if you have issues, I can walk you through restoring that way
